presto-main/src/main/java/com/facebook/presto/sql/gen/SqlTypeBytecodeExpression.java

```diff
@@ -98,7 +98,7 @@ public BytecodeExpression writeValue(BytecodeExpression blockBuilder, BytecodeEx
             return invoke("writeBoolean", void.class, blockBuilder, value);
         }
         if (fromJavaElementType == long.class) {
-            return invoke("writeLong", void.class, blockBuilder, value);
+            return invoke("writeLong", void.class, blockBuilder, value.cast(long.class));
```

Comment:
I added this because of a test failure in TestStateCompiler. I think byte also uses this code path but it would complain
java.lang.NoSuchMethodError: com.facebook.presto.spi.type.Type.writeLong(Lcom/facebook/presto/spi/block/BlockBuilder;B)V